pkg/resource/resolver_endpoint/sdk.go

```diff
@@ -0,0 +1,633 @@
+	ko := desired.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	if resp.ResolverEndpoint.Arn != nil {
+		arn := ackv1alpha1.AWSResourceName(*resp.ResolverEndpoint.Arn)
+		ko.Status.ACKResourceMetadata.ARN = &arn
+	}
+	if resp.ResolverEndpoint.CreationTime != nil {
+		ko.Status.CreationTime = resp.ResolverEndpoint.CreationTime
+	} else {
+		ko.Status.CreationTime = nil
+	}
+	if resp.ResolverEndpoint.CreatorRequestId != nil {
+		ko.Status.CreatorRequestID = resp.ResolverEndpoint.CreatorRequestId
+	} else {
+		ko.Status.CreatorRequestID = nil
+	}
+	if resp.ResolverEndpoint.Direction != nil {
+		ko.Spec.Direction = resp.ResolverEndpoint.Direction
+	} else {
+		ko.Spec.Direction = nil
+	}
+	if resp.ResolverEndpoint.HostVPCId != nil {
+		ko.Status.HostVPCID = resp.ResolverEndpoint.HostVPCId
+	} else {
+		ko.Status.HostVPCID = nil
+	}
+	if resp.ResolverEndpoint.Id != nil {
+		ko.Status.ID = resp.ResolverEndpoint.Id
+	} else {
+		ko.Status.ID = nil
+	}
+	if resp.ResolverEndpoint.IpAddressCount != nil {
+		ko.Status.IPAddressCount = resp.ResolverEndpoint.IpAddressCount
+	} else {
+		ko.Status.IPAddressCount = nil
+	}
+	if resp.ResolverEndpoint.ModificationTime != nil {
+		ko.Status.ModificationTime = resp.ResolverEndpoint.ModificationTime
+	} else {
+		ko.Status.ModificationTime = nil
+	}
+	if resp.ResolverEndpoint.Name != nil {
+		ko.Spec.Name = resp.ResolverEndpoint.Name
+	} else {
+		ko.Spec.Name = nil
+	}
+	if resp.ResolverEndpoint.ResolverEndpointType != nil {
+		ko.Spec.ResolverEndpointType = resp.ResolverEndpoint.ResolverEndpointType
+	} else {
+		ko.Spec.ResolverEndpointType = nil
+	}
+	if resp.ResolverEndpoint.SecurityGroupIds != nil {
+		f10 := []*string{}
+		for _, f10iter := range resp.ResolverEndpoint.SecurityGroupIds {
+			var f10elem string
+			f10elem = *f10iter
+			f10 = append(f10, &f10elem)
+		}
+		ko.Spec.SecurityGroupIDs = f10
+	} else {
+		ko.Spec.SecurityGroupIDs = nil
+	}
+	if resp.ResolverEndpoint.Status != nil {
+		ko.Status.Status = resp.ResolverEndpoint.Status
+	} else {
+		ko.Status.Status = nil
+	}
+	if resp.ResolverEndpoint.StatusMessage != nil {
+		ko.Status.StatusMessage = resp.ResolverEndpoint.StatusMessage
+	} else {
+		ko.Status.StatusMessage = nil
+	}
+
+	rm.setStatusDefaults(ko)
+	return &resource{ko}, nil
+}
+
+// newUpdateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Update API call for the resource
+func (rm *resourceManager) newUpdateRequestPayload(
+	ctx context.Context,
+	r *resource,
+	delta *ackcompare.Delta,
+) (*svcsdk.UpdateResolverEndpointInput, error) {
+	res := &svcsdk.UpdateResolverEndpointInput{}
+
+	if r.ko.Spec.Name != nil {
+		res.SetName(*r.ko.Spec.Name)
+	}
+	if r.ko.Status.ID != nil {
+		res.SetResolverEndpointId(*r.ko.Status.ID)
+	}
+	if r.ko.Spec.ResolverEndpointType != nil {
+		res.SetResolverEndpointType(*r.ko.Spec.ResolverEndpointType)
+	}
+
```

Comment:
To complete full support for updates, we ideally want to inject new IPAddresses here.. but it's fine if we do that in a future PR.